### PR TITLE
React task 7

### PR DESCRIPTION
**Pull request by AlexandraBeseda**
- added and changed logic in Cart component;
- created page OrderHistory;
- refactor Cart and all children components;
- changed logic in thunk deleteAccount (specifically localStorage).

In this task I added OrdersHistory page. If you are new user your OrderHistory would be empty.  
image
image
If you create new order and submit it, your OrderHistory will have positions.
image
image
image

### CODE REVIEW

File: application/src/bll/reducers/paymentCardReducer.ts [hunk 1]
```diff
@@ -54,4 +54,30 @@ export const setCustomerData = (
...
+  name: string,
+  mobile: string,
+  totalCartSum: number
+) => {
+  const email = localStorage.getItem('email');
+  const time = new Date();
+  const day = time.getUTCDate();
+  const month = time.getMonth() + 1;
+  const year = time.getFullYear();
+  const date = `${day}.${month}.${year}`;
```
Comment (dpryshchep) [inline application/src/bll/reducers/paymentCardReducer.ts]: Let's create common util function which will be responsible for data treatment

File: application/src/pages/Cart/CarInCart/CarInCart.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const dispatch = useDispatch();
+
+  const setNewAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    dispatch(addAmount(props.id, Number(e.currentTarget.value)));
+  };
+
+  const handleDeleteCar = () => {
+    dispatch(deleteCar(props.id));
+  };
+  const total = Math.floor(props.total * 100) / 100;
```
Comment (dpryshchep) [inline application/src/pages/Cart/CarInCart/CarInCart.tsx]: You have duplication of Math.floor for total calculation
let's create common util function

```
formatTotalAmount(props.total)
...
const formatTotalAmount = (amount) => {
...
}
```

File: application/src/pages/Cart/Cart.tsx [hunk 3]
```diff
@@ -1,55 +1,44 @@
...
-  }
 
   let cartTotalSum = 0;
   cartReducer.forEach((car) => {
     cartTotalSum += car.total;
   });
 
+  const totalCarPrice = Math.floor(cartTotalSum * 100) / 100;
+
+  if (cartReducer.length < 1) {
```
Comment (dpryshchep) [inline application/src/pages/Cart/Cart.tsx]: It's hard to understand what is role for that condition, so it's better

```
const isCartEmpty = cartReducer.length < 1;

if (isCartEmpty) {
... your code here
}
```

File: application/src/pages/Cart/Cart.tsx [hunk 4]
```diff
@@ -1,55 +1,44 @@
...
+    return <EmptyCart />;
+  }
+  if (cartReducer.length > 0 && paymentCarDReducerTotal > 0) {
```
Comment (dpryshchep) [inline application/src/pages/Cart/Cart.tsx]: Same as above, it's necessary to give some name to the condition

File: application/src/pages/Cart/CartForm/CartFormHeader/CartFormHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+export const CartFormHeader: React.FC<CartFormPropTypes> = ({
+  orderNum,
+  total
+}) => {
+  const { t } = useTranslation();
+  return (
+    <div className={style.title}>
+      <p>{`${t('cart.order.totalAmount')} ${total} $`}</p>
+      <div>
+        {orderNum.length > 1 &&
```
Comment (dpryshchep) [inline application/src/pages/Cart/CartForm/CartFormHeader/CartFormHeader.tsx]: Please create variable for `orderNum.length > 1`

Comment (dpryshchep) [reply application/src/pages/Cart/CartForm/CartFormHeader/CartFormHeader.tsx]: What is 1?

File: application/src/pages/Cart/CartForm/CustomForm.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            placeholder={t('cart.form.mobile')}
+            type="mobile"
+            name="mobile"
+            onChange={handleChange}
+            onBlur={handleBlur}
+            value={values.mobile}
+          />
+          <div className={style.error}>
+            {errors.mobile &&
+              touched.mobile &&
```
Comment (dpryshchep) [inline application/src/pages/Cart/CartForm/CustomForm.tsx:106]: it's better to create variable for this condition like
`const isMobileNumberValid = ...`

File: application/src/bll/reducers/paymentCardReducer.ts (hunk 1 above)
Comment (AlexandraBeseda) [reply application/src/bll/reducers/paymentCardReducer.ts]: I created function getDate(total:number){...}

File: application/src/pages/Cart/CarInCart/CarInCart.tsx (hunk 2 above)
Comment (AlexandraBeseda) [reply application/src/pages/Cart/CarInCart/CarInCart.tsx]: done

File: application/src/pages/Cart/Cart.tsx (hunk 3 above)
Comment (AlexandraBeseda) [reply application/src/pages/Cart/Cart.tsx]: fixed

File: application/src/pages/Cart/Cart.tsx (hunk 4 above)
Comment (AlexandraBeseda) [reply application/src/pages/Cart/Cart.tsx]: fixed

File: application/src/pages/Cart/CartForm/CartFormHeader/CartFormHeader.tsx (hunk 5 above)
Comment (AlexandraBeseda) [reply application/src/pages/Cart/CartForm/CartFormHeader/CartFormHeader.tsx]: it was my mistake, i fixed and refactor

File: application/src/pages/Cart/CartForm/CustomForm.tsx (hunk 6 above)
Comment (AlexandraBeseda) [reply application/src/pages/Cart/CartForm/CustomForm.tsx:106]: I think to create here variable is unnessary, because it has repeated one time

**Pull request merged**